ste/downloader-azureFiles_windows.go

```diff
@@ -147,28 +148,34 @@ func (a *azureFilesDownloader) PutSDDL(sip ISMBPropertyBearingSourceInfoProvider
 	isProtectedAtSource := (ctl & windows.SE_DACL_PROTECTED) != 0
 	isAtTransferRoot := len(splitPath) == 1
 
-	if isProtectedAtSource || isAtTransferRoot || a.parentIsShareRoot(txInfo.Source) {
-		securityInfoFlags |= windows.PROTECTED_DACL_SECURITY_INFORMATION
+	parsedSDDL, err := sddl.ParseSDDL(sddlString)
+
+	if err != nil {
+		panic(fmt.Sprintf("Sanity check; SDDL failed to parse (downloader-azureFiles_windows.go), %s", err)) // We already parsed it. This is impossible.
```

Comment:
QQ: Why are we not returning error and panicking instead? Does unable to parse SDDL mean we have to fail immediately fatal?

Reply:
The SDDL has successfully been parsed in the past-- it doesn't make sense for it to fail here, so, if it has, something is _severely_ wrong.